src/Stratis.Bitcoin/Consensus/ConsensusManager.cs

```diff
@@ -1250,6 +1251,23 @@ public ChainedHeaderBlock GetBlockData(uint256 blockHash)
             return chainedHeaderBlock;
         }
 
+        /// <inheritdoc />
+        public ChainedHeaderBlock[] GetBlockData(List<uint256> blockHashes)
+        {
+            Dictionary<uint256, ChainedHeaderBlock> chainedHeaderBlocks;
+
+            lock (this.peerLock)
+            {
+                chainedHeaderBlocks = this.chainedHeaderTree.GetChainedHeaderBlocks(blockHashes).ToDictionary(chb => chb.ChainedHeader.HashBlock, chb => chb);
+            }
+
+            List<uint256> getFromStore = chainedHeaderBlocks.Where(kv => kv.Value != null && kv.Value.Block == null).Select(kv => kv.Key).ToList();
```

Comment:
Can you please make this code more readable? less linq magic perhaps. the equivilant single block method is far more easier to read.

Reply:
Done.